test/swar/BasicOperations.cpp

```diff
@@ -565,10 +583,188 @@ TEST_CASE(
     "saturatingUnsignedAddition",
     "[swar][saturation]"
 ) {
-    CHECK(SWAR<4, u16>(0x0200).value() == saturatingUnsignedAddition(SWAR<4, u16>(0x0100), SWAR<4, u16>(0x0100)).value());
-    CHECK(SWAR<4, u16>(0x0400).value() == saturatingUnsignedAddition(SWAR<4, u16>(0x0100), SWAR<4, u16>(0x0300)).value());
-    CHECK(SWAR<4, u16>(0x0B00).value() == saturatingUnsignedAddition(SWAR<4, u16>(0x0800), SWAR<4, u16>(0x0300)).value());
-    CHECK(SWAR<4, u16>(0x0F00).value() == saturatingUnsignedAddition(SWAR<4, u16>(0x0800), SWAR<4, u16>(0x0700)).value());
-    CHECK(SWAR<4, u16>(0x0F00).value() == saturatingUnsignedAddition(SWAR<4, u16>(0x0800), SWAR<4, u16>(0x0800)).value());
-    CHECK(S4_32(0x0F0C'F000).value() == saturatingUnsignedAddition(S4_32(0x0804'F000), S4_32(0x0808'F000)).value());
+    CHECK(S4_16(0x0200).value() == saturatingUnsignedAddition(S4_16(0x0100), S4_16(0x0100)).value());
+    CHECK(S4_16(0x0400).value() == saturatingUnsignedAddition(S4_16(0x0100), S4_16(0x0300)).value());
+    CHECK(S4_16(0x0B00).value() == saturatingUnsignedAddition(S4_16(0x0800), S4_16(0x0300)).value());
+    CHECK(S4_16(0x0F00).value() == saturatingUnsignedAddition(S4_16(0x0800), S4_16(0x0700)).value());
+    CHECK(S4_16(0x0F00).value() == saturatingUnsignedAddition(S4_16(0x0800), S4_16(0x0800)).value()); 
+    CHECK(S4_32(0x0F0C'F000).value() == saturatingUnsignedAddition(S4_32(0x0804'F000), S4_32(0x0808'F000)).value()); 
+}
+
+TEST_CASE(
+    "cheapNullChecking",
+    "[swar]"
+) {
+    {
+        // The last value / lsb null lane is all these care about, but the
+        // implementation is so tied to the execution we check both.
+        CHECK(S4_32(0x0008'0080).value() == firstZeroLane<4, uint32_t>( S4_32(0xaaa0'aa0a)));
+        CHECK(1 == S4_32(firstZeroLane<4, uint32_t>( S4_32(0xaaa0'aa0a))).lsbIndex());
+        CHECK(S4_32(0x8888'8888).value() == firstZeroLane<4, uint32_t>( S4_32(0x1110'1110)));
+        CHECK(S4_32(0x8800'0000).value() == firstZeroLane<4, uint32_t>( S4_32(0x1025'3552)));
+        CHECK(S4_32(0x8880'0000).value() == firstZeroLane<4, uint32_t>( S4_32(0x1105'5352)));
+
```

Comment:
These checks are unacceptable, because we use the new concept of pseudo boolean SWAR, the proper checks would treat the upper bits/lanes as "don't cares".

Reply:
When we codify pseudoBooleanSWAR a little better, I'll be happy to do that.
For now, I'd prefer to leave the brittle implementation specific checks in because it matches what exists and breaks if it changes for some unexpected reason.
If you really feel strongly I can post-mask, I don't have that strong of an opinion on it.

Reply:
I have consistently missed the opportunity to show how to use the constructs in this library of SWAR in the unit tests.  I am trying to revert that negative inertia.
Please try to illustrate the "pseudoness" in these unit tests, precisely because it is an emergent concept: if the "post-masking" becomes annoying, then perhaps we can further articulate the "pseudoness" by ergonomics for it.

Reply:
Added a small example, can improve that over time.